ext/standard/user_filters.c

```diff
@@ -384,38 +386,42 @@ PHP_FUNCTION(stream_bucket_make_writeable)
 static void php_stream_bucket_attach(int append, INTERNAL_FUNCTION_PARAMETERS)
 {
 	zval *zbrigade, *zobject;
-	zval *pzbucket, *pzdata;
+	zval *pzbucket, *pzdata, rv;
 	php_stream_bucket_brigade *brigade;
 	php_stream_bucket *bucket;
 
 	ZEND_PARSE_PARAMETERS_START(2, 2)
 		Z_PARAM_RESOURCE(zbrigade)
-		Z_PARAM_OBJECT(zobject)
+		Z_PARAM_OBJECT_OF_CLASS(zobject, stream_bucket_class_entry)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (NULL == (pzbucket = zend_hash_str_find_deref(Z_OBJPROP_P(zobject), "bucket", sizeof("bucket")-1))) {
-		zend_argument_value_error(2, "must be an object that has a \"bucket\" property");
+	if ((brigade = (php_stream_bucket_brigade*)zend_fetch_resource(
+		Z_RES_P(zbrigade), PHP_STREAM_BRIGADE_RES_NAME, le_bucket_brigade)) == NULL) {
 		RETURN_THROWS();
 	}
 
-	if ((brigade = (php_stream_bucket_brigade*)zend_fetch_resource(
-					Z_RES_P(zbrigade), PHP_STREAM_BRIGADE_RES_NAME, le_bucket_brigade)) == NULL) {
+	if (NULL == (pzbucket = zend_read_property(NULL, Z_OBJ_P(zobject), "bucket", sizeof("bucket")-1, false, &rv))) {
+		zend_argument_value_error(2, "must be an object that has a \"bucket\" property");
```

Comment:
any reason why you swapped the check?

Reply:
I mean before bucket was checked first and then brigade fetched...

Reply:
it will unnecessary result in different error if both bucket and brigade are invalid. Although it might not matter that much once brigade becomes object...

Reply:
I made this change because
- the former check used to validate the latter argument and vice versa. In my opinion, it's a good practice to validate the arguments in order
- after this change, the bucket property read and the extraction of its steam are directly after each other. Previously, the extraction of the brigade resource separated them.

Yeah, I think the change of the error message should not be a big problem. Either way, users have to solve both issues...